### DISCUSSION TITLE
Has the equivalent of scala.reflect.macros.Parsers been removed?

### DISCUSSION THREAD

**Discussion by giiita** [scala/scala3#12500] [Metaprogramming] [answered]
In Scala2's macro, I could dynamically construct expressions by using Context.parse and so on.
For example:

```scala
  def generateExpr[T: c.WeakTypeTag](target: C#Symbol): C#Expr[T] = {
    if (target.isModule) {
      c.Expr[T](c.parse(target.fullName))
    } else {
      c.Expr[T] {
        c.parse(
          s"""new ${target.fullName}(${
            val a = target.asClass.primaryConstructor.asMethod
            val b = a.paramLists
            target.asClass.primaryConstructor.asMethod.paramLists
              .collect {
                case curry if !curry.exists(_.isImplicit) =>
                  curry.map { param => s"inject[${param.typeSignature.toString}]" }.mkString(",")
              }
              .mkString(")(")
          }) with MixIn"""
        )
      }
    }
  }
```

I was able to expand the TypeSignature into a Quasiquotes.
This may have been an experimental feature, but it seems to be an important one at times.
In fact, I saw so many OSS written using this feature. And I was one of those who relied on this feature.

It is not easy to define an EXpression that takes `[T: Type]` and returns `new T(...) with MixinTraits`. I have no idea.
Removing this feature may make migration more difficult.

**Comment by giiita** [comment, marked as answer]
> Those are use cases that we explicitly did not want to support since they cause language fragmentation, with many possible dialects supported by macros. I am sure there are good use cases
for this, but overall I believe allowing this would be detrimental to the language ecosystem as a whole.

https://contributors.scala-lang.org/t/compatibility-required-for-migration-from-scala2-macro/5100/2?u=giiita

**Reply by nicolasstucki** [reply]
`parse` is not supported and will not be added. The ability to create arbitrary classes will be added to the reflection interface.
